plugins/scaleflux/sfx-nvme.c

```diff
@@ -594,6 +602,143 @@ static int sfx_get_bad_block(int argc, char **argv, struct command *cmd, struct
 	return 0;
 }
 
+static __u64 get_capacity(int fd, sfx_capacity_ctx *cap_val)
+{
+	struct sfx_freespace_ctx ctx = { 0 };
+	__u64 cap_GB = 0;
+	__u64 pcap_GB = 0;
+	if (ioctl(fd, SFX_GET_FREESPACE, &ctx)) {
+		fprintf(stderr, "Get p_capacity fail, errno=%d\n", errno);
+		return INVALID_PARAM;
+	}
+	/*Convert to GB*/
+	cap_GB = IDEMA_CAP2GB(ctx.user_space);
+	pcap_GB = ((ctx.phy_space - 97696368llu) / 1953504llu) + 50llu;
+
+	cap_val->capacity = cap_GB;
+	cap_val->p_capacity = pcap_GB;
+	return 0;
+}
+
+/**
+ * @brief convert the file name to it's target file name if the input file is symbol link.
+ * Otherwise, just copy the input file name to target file name
+ *
+ * @param dir		abs dir path of input file
+ * @param link_name	input file name
+ * @param tg_name	target file name
+ * @param tg_size	size of tg_name
+ *
+ * @return 0, success; -1, fail
+ */
+static int name_from_link(const char *dir, char *link_name, char *tg_name,
+			  int tg_size)
+{
+	char path[50] = { 0 };
+	struct stat f_state;
+
+	if (!dir || !link_name || !tg_name || tg_size == 0) {
+		fprintf(stderr, "%s: Invalid params\r\n", __func__);
+		return -1;
+	}
+	memset(tg_name, 0x00, tg_size);
+	sprintf(path, "%s/%s", dir, link_name);
+
+	if (lstat(path, &f_state) < 0) {
+		fprintf(stderr, "lstate fail, errno=%d\r\n", errno);
+		return -1;
+	}
+	if (S_ISLNK(f_state.st_mode)) {
+		if (readlink(path, tg_name, tg_size) < 0) {
+			fprintf(stderr, "readlink fail, errno =%d \n", errno);
+			return -1;
+		}
+	} else {
+		snprintf(tg_name, tg_size, "%s", link_name);
+	}
+	return 0;
+}
+
+static int nvme_block_from_char(char *char_dev, char *blk_dev)
```

Comment:
This doesn't actually work with the linux driver. And there should already be a function that does this, if I'm not mistaken.

Reply:
The prototype is from nvme_block_from_char() but do convert scaleflux misc char device name "sfxv*" to block device name "sfdv*n1". I will change the function to avoid confusion.

Reply:
Do you have any idea about why the pull request show diffs which already in master?